src/signatures/secp256k1_ecdsa.rs

```diff
@@ -0,0 +1,210 @@
+// Copyright 2023 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use core::convert::TryFrom;
+use core::hash::{Hash, Hasher};
+
+use zeroize::ZeroizeOnDrop;
+
+#[derive(Clone, Copy, Debug, Eq, Ord, PartialEq, PartialOrd)]
+pub struct Address([u8; Self::LENGTH]);
```

Comment:
Should this really be part of crypto? Is it part of the specs? I feel like most of the time it would be down to the consumer to decide how to derive the address from the public key?

Reply:
Where should it be? I understand that it's not directly a part of Secp256k1/ECDSA spec but the whole point was to generate EVM addresses. I haven't found the spec for it, but here's the code that does that https://github.com/gakonst/ethers-rs/blob/f7454f27666750449ca9005a4db032025b4c36ff/ethers-signers/src/aws/utils.rs#L59

Reply:
I mean, I assume how you want to derive an address from a public key could change from one project to another. Which is similar to what we do atm with ED25519, crypto.rs provides us a public key and we hash it with blake2 on our own to derive the address from it. This part is not done in crypto.rs.

Reply:
Fair enough. So where should it be? In stronghold? If it's outside of crypto then we need to expose uncompressed public key bytes.

Reply:
Stronghold itself doesn't really need an address since it only delivers the public key to the consumer. For ED25519 we derive the address in the SDK directly so I guess we should do the same? https://github.com/iotaledger/iota-sdk/blob/develop/sdk/src/client/stronghold/secret.rs#L87